### PR TITLE
Feat/incommon updates

### PR DESCRIPTION
**Pull request by paulineribeyre**
The first commit copies the changes from release 2.7.2 to master (InCommon updates)
Other commits:
- Shibboleth documentation
- add `idp` and `shib_idp` query parameters to `/login/fence` endpoint. Previously they were only in `/authorize` endpoint
- Improve the `ENABLED_IDENTITY_PROVIDERS` config format to allow enabling several login options for the same provider

TODO:
- [x] update docs

### New Features
- Support InCommon login: add "idp" and "shib_idp" query parameters to "/login/shib", "/login/fence" and "/authorize"

### Deployment changes
- Fence config ENABLED_IDENTITY_PROVIDERS: "providers" field deprecated and replaced by "login_options" to allow enabling several login options for the same provider (backwards compatible)

### CODE REVIEW

File: fence/blueprints/login/__init__.py [hunk 1]
```diff
@@ -71,72 +88,190 @@ def default_login():
...
+
+def get_all_shib_idps():
+    """
+    Get the list of all existing Shibboleth IDPs.
+    This function only returns the information we need to generate login URLs.
+
+    Returns:
+        list: list of {"idp": "", "name": ""} dictionaries
+    """
+    res = requests.get("https://login.bionimbus.org/Shibboleth.sso/DiscoFeed")
```
Comment (fantix) [inline fence/blueprints/login/__init__.py]: in config 😏

File: docs/shibboleth.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# Shibboleth / InCommon login
+
+Shibboleth is set up at login.bionimbus.org and lets us log in through InCommon by specifying the `shib_idp` parameter (as of fence release 4.7.0 and fence-shib release 2.7.2). If no `shib_idp` is specified (or if using an earlier fence version), users will be redirected to the NIH login page by default.
```
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply docs/shibboleth.md]: yes i think so, for backwards compatibility

File: docs/shibboleth.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+-> redirect to portal
+```
+
+## Configuration
+
+### In login.bionimbus.org
+
+The [Shibboleth dockerfile](../DockerfileShib) image is at https://quay.io/repository/cdis/fence-shib and is NOT compatible yet with python 3/the latest fence (for now, use fence 2.7.x).
+
+`login.bionimbus.org` is under the Genomel AWS acccount. The deployment only includes `revproxy` and `fenceshib`. The fence configuration enables the `shibboleth` provider:
```
[the remark this answers is not in the document]
Comment (paulineribeyre) [reply docs/shibboleth.md]: moving the doc to the wiki

**Comment by paulineribeyre** [comment]
merging with failed 1 failed check - we know that the only failed jenkins test, `ExportToWorkspaceTest: Export default manifest, mount it and check manifest name`, has been flaky recently

**Pull request merged**